Rakefile.rb

```diff
@@ -48,6 +48,36 @@ def name_from(json)
   orgs.first[:name]
 end
 
+def file_to_commit_metadata
+  # There's about 2MB of data returned by this command, so rather than
+  # using backticks to turning it into a string and then splitting it,
+  # parse the output line-by-line. As a bonus this doesn't
+  # unnecessarily invoke a shell.
+  command = ['git', '--no-pager', 'log', '--name-only',  '--format=%h|%at', '--', 'data']
+  result = {}
+  sha, timestamp = nil
+  Kernel.open('|-', 'r') { |f|
+    if f
+      # In the parent process after forking, parse the output:
+      f.each_line { |line|
+        commit_match = line.match /^([a-f\d]+)\|(\d+)$/
+        if commit_match
+          sha, timestamp = commit_match.captures
+        else
+          filename = line.strip
+          unless result.has_key? filename
+            result[filename] = {sha: sha, timestamp: timestamp}
+          end
```

Comment:
I think this would be clearer, and more Rubyish as simply
```ruby
  result[filename] ||= { sha: sha, timestamp: timestamp } 
```
I'd probably drop the temporary `filename` variable too:
```ruby
else
  result[line.strip] ||= { sha: sha, timestamp: timestamp }
end
```
but that's less clear-cut, as at the minute it helps document the input format a little. I suspect, though, it might be better still to make it clearer at a higher level what that format is, as that's the part I'm finding it hard to grok initially.